Framework/MDAlgorithms/src/IntegratePeaksMD2.cpp

```diff
@@ -411,28 +414,21 @@ void IntegratePeaksMD2::integrate(typename MDEventWorkspace<MDE, nd>::sptr ws) {
                                       static_cast<coord_t>(PeakRadius),
                                       static_cast<coord_t>(cylinderLength),
                                       signal, errorSquared, signal_fit);
-      for (size_t j = 0; j < numSteps; j++) {
-        wsProfile2D->dataX(i)[j] = static_cast<double>(j);
-        wsProfile2D->dataY(i)[j] = signal_fit[j];
-        wsProfile2D->dataE(i)[j] = std::sqrt(signal_fit[j]);
-      }
 
       // Integrate around the background radius
       if (BackgroundOuterRadius > PeakRadius) {
         // Get the total signal inside "BackgroundOuterRadius"
 
         signal_fit.clear();
-        for (size_t j = 0; j < numSteps; j++)
-          signal_fit.push_back(0.0);
+        std::fill(signal_fit.begin(), signal_fit.end(), 0.0);
```

Comment:
Minor comment. If you used `Counts` instead of `std::vector` for signal_fit, you could just use `signal_fit = 0;`